mlflow/pyfunc/__init__.py

```diff
@@ -24,14 +24,79 @@
 <mlflow.pyfunc.PyFuncModel.predict>` method, which has the following signature::
 
   predict(
-    model_input: [pandas.DataFrame, numpy.ndarray, scipy.sparse.(csc.csc_matrix | csr.csr_matrix),
+    model_input: [pandas.DataFrame, numpy.ndarray, scipy.sparse.(csc_matrix | csr_matrix),
     List[Any], Dict[str, Any], pyspark.sql.DataFrame]
-  ) -> [numpy.ndarray | pandas.(Series | DataFrame) | List | pyspark.sql.DataFrame]
+  ) -> [numpy.ndarray | pandas.(Series | DataFrame) | List | Dict | pyspark.sql.DataFrame]
 
 All PyFunc models will support `pandas.DataFrame` as input and PyFunc deep learning models will
 also support tensor inputs in the form of Dict[str, numpy.ndarray] (named tensors) and
 `numpy.ndarrays` (unnamed tensors).
 
+Here are some examples of supported inference types, assuming we have the correct ``model`` object
+loaded.
+
+.. list-table::
+    :widths: 30 70
+    :header-rows: 1
+    :class: wrap-table
+
+    * - Input Type
+      - Example
+    * - ``pandas.DataFrame``
+      -
+        .. code-block:: python
+
+            import pandas as pd
+
+            x_new = pd.DataFrame(dict(x1=[1,2,3], x2=[4,5,6]))
```

Comment:
super-nit: Is there any particular reason for saying `_new`? Just a personal impression, but if I find `x_new`, my brain start to search sth like `x` or `x_old` as a counterpart.

Reply:
To me, `x_new` implies any inference data e.g. new data unseen by the model. I've also seen this used in many blogs/tutorials, specifically for sklearn - [example](https://machinelearningmastery.com/make-predictions-scikit-learn/). 

Let me know if you think we should rename.

Reply:
Oh I see, tbh I've never seen this (probably because I came from DL side - not really reading through sklearn tutorials). Nevertheless there IS a practice, so the current name is totally fine. Thanks for the clarification!